### PR TITLE
Introduce useGet and useMutate hooks

### PR DESCRIPTION
**Pull request by brmscheiner**

### CODE REVIEW

File: src/hooks/useMutate.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  useEffect(
+    () => {
+      if (error) {
+        setDisplayedError(error);
+      }
+    },
+    [error],
+  );
+
+  const mutate = mutationArgs => mutation.mutateAsync(mutationArgs);
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/hooks/useMutate.js:72]: i think async will always be okay. but we could also provide something like 

``` const mutateSync = mutationArgs => mutation.mutate(mutationArgs) ```

for that special case where it's not

File: src/models/missions/usePatchMission.js [hunk 2]
```diff
@@ -1,38 +1,15 @@
...
-    if (result?.status === 200) {
-      queryClient.invalidateQueries(getMissionQueryKey(missionId));
-      queryClient.invalidateQueries(queryKeys.me);
-    }
-
-    return result;
+  return usePatch({
+    deriveUrl: ({ missionGuid }) => `/missions/${missionGuid}`,
+    deriveData: ({ operations }) => operations,
+    deriveQueryKeys: ({ missionGuid }) => [
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/models/missions/usePatchMission.js:10]: two keys!

File: src/hooks/useMutate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Atticus29) [reply src/hooks/useMutate.js:72]: Ok. I support this.

File: src/models/missions/usePatchMission.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Atticus29) [reply src/models/missions/usePatchMission.js:10]: Nice. Good.

**Pull request merged**